### DISCUSSION TITLE
Integrate Proximity which enables preheating when travelling home.

### DISCUSSION THREAD

**Discussion by mdehaas** [jmcollin78/versatile_thermostat#378] [Ideas]
**High Level description**
I recently became aware of the [Proximity integration](https://www.home-assistant.io/integrations/proximity/) in Home Assistant. The documentation mentions the use-case to influence a thermostat based on travel direction (and distance) to and from home. This can be accomplished by writing custom automations but I think this could be a valuable addition to Versatile Thermostat. For example Tado° offers this functionality as part of their subscription plan.

**Describe the solution you'd like**
Have the option to select a Proximity Device to monitor and be able to define a distance for both moving towards and away for that device to which VT switches the between the same preset (Eco, Boost, etc.) but switches between Home and Away temperature presets, or two more states van be introduced (i.e. Arriving/Leaving).
The implementation of the Proximity integration allows for two main modes of operation (which can be mixed): device_tracker and user platform based. Sensor are created of which i feel 'Direction of travel' and 'Distance' are the most valuable for this functionality. The 'Nearest...' sensor do not make sense as I'm using seperate Proximity devices, each representing one user entity. The various configurations need to be discussed (As in seperate rooms controlled by one user entity and shared rooms (livingroom) controlled by multiple proximity services or one proximity service representing multiple users). This discussion should be input for the documentation and 'best practices' to configure this.
The proximity platform is fed by any device tracker and updates with the same frequency (default for HA Companion app is 5 minutes if I'm correct.)

Please give input and share your usage of the Proximity integration.

Kind regards,
Marcel de Haas

**Comment by maia** [comment]
I would suggest creating a binary helper sensor "at home or on the way home" and an automation which toggles this device based on the data from the proximity sensor. Then you could use this binary sensor (or multiple, one for each person) as presence state for VTherm.

**Reply by mdehaas** [reply]
This is a viable solution but this also requires custom automations. Having this functionality which is tightly coupled with the operation of a thermostat in one place and be user-friendly to people with less experience in HA automations makes sense to me. Automations have a place to bind together stuff which is implementation specific (like turn the thermostat of when all lights are turned of). I feel this functionality can used without customization (just configuration) and therefore the place is in VT.

**Reply by maia** [reply]
I understand. But what I really miss in the HA ecosystem are modules that can be maintained once but used by multiple integrations/blueprints/… – this is a perfect example, as others might want to do other things when someone is close to back home (turn on outdoor lights, open the garage door,…). I do not think that VTherm is *the* place this functionality should be provided, probably it's rather a "proximity deluxe" sensor that can be installed and then be used by VTherm and others.

Another example (but off topic) are time based moving averages. So many integrations could use a proper implementation but instead they all have to roll their own (and in the VTherm example, the EMA by VTherm is worse than the [average sensor platform](https://github.com/Limych/ha-average) provides).
